mllib/src/main/scala/org/apache/spark/mllib/util/LinearDataGenerator.scala

```diff
@@ -112,8 +139,13 @@ object LinearDataGenerator {
     x.foreach { v =>
       var i = 0
       val len = v.length
+      val sparceRnd = new Random(seed)
```

Comment:
If we use same random generator for both creating features and choice which columns to zero, x is  different from current ones. This cause unit test failures. Can we change the assertion tolerance or target written in `LinearRegressionSuite`?